Engine/Common/Types.h

```diff
@@ -126,290 +126,557 @@ struct StartEndIndex {
+    // ------------------ Functions ------------------ //
+    // getters //
+    DLLEXPORT constexpr int GetX() const noexcept
+    {
+        return X;
+    }
 
-    DLLEXPORT void SetData(const int& data)
+    DLLEXPORT constexpr int GetY() const noexcept
     {
-        X = data;
-        Y = data;
-    };
-    DLLEXPORT void SetData(const int& data1, const int& data2)
+        return Y;
+    }
+
+    // setters //
+    DLLEXPORT inline void SetX(int val)
     {
-        X = data1;
-        Y = data2;
-    };
+        X = val;
+    }
 
-    VALUE_TYPE(Int2);
+    DLLEXPORT inline void SetY(int val)
+    {
+        Y = val;
+    }
 
-    int X, Y;
-};
+    // add all elements together //
+    DLLEXPORT constexpr int HAdd() const noexcept
+    {
+        return X + Y;
+    }
 
-struct Int3 {
-public:
-    DLLEXPORT Int3()
+    // Add all elements together after abs() is called on each element //
+    DLLEXPORT inline unsigned int HAddAbs() const noexcept
+    {
+        return std::abs(X) + std::abs(Y);
+    }
+
+    // getting min and max of objects //
+    DLLEXPORT constexpr Int2 MinElements(const Int2& other) const noexcept
+    {
+        return Int2(X < other.X ? X : other.X, Y < other.Y ? Y : other.Y);
+    }
+
+    DLLEXPORT constexpr Int2 MaxElements(const Int2& other) const noexcept
+    {
+        return Int2(X > other.X ? X : other.X, Y > other.Y ? Y : other.Y);
+    }
+
+    // value clamping //
+    DLLEXPORT constexpr Int2 Clamp(const Int2& min, const Int2& max) const noexcept
+    {
+        const Int2 minval = this->MinElements(max);
+        return min.MaxElements(minval);
+    }
+
+    // ----------------- Vector math ------------------- //
+    // dot product of the vectors //
+    DLLEXPORT constexpr int Dot(const Int2& val) const noexcept
     {
-        X = 0;
-        Y = 0;
-        Z = 0;
+        return X * val.X + Y * val.Y;
     }
-    DLLEXPORT Int3(int x, int y, int z)
+
+    // length of the vector //
+    DLLEXPORT inline float Length() const noexcept
     {
-        X = x;
-        Y = y;
-        Z = z;
+        return static_cast<float>(sqrt(X * X + Y * Y));
     }
-    DLLEXPORT explicit Int3(int data)
+
+    DLLEXPORT constexpr unsigned int LengthSquared() const noexcept
     {
-        // save a bit of space //
-        X = Y = Z = data;
+        return X * X + Y * Y;
     }
 
     // ------------------------------------ //
-    DLLEXPORT Int3 operator+(const Int3& val) const
+    // static returns //
+    // creates a Int2 with all zeros //
+    DLLEXPORT constexpr static Int2 zero() noexcept
     {
-        return Int3(X + val.X, Y + val.Y, Z + val.Z);
+        return Int2(0);
     }
 
-    DLLEXPORT Int3 operator*(int val) const
+    // creates a Int2 with all ones //
+    DLLEXPORT constexpr static Int2 one() noexcept
     {
-        return Int3(X * val, Y * val, Z * val);
+        return Int2(1);
+    }
+
+    // unitary vector x, to work with ozz declarations //
```

Comment:
These old `*_axis` things are very likely unused and should probably be removed along with the note about ozz declarations.